api/python/provisioner/commands/deploy_vm.py

```diff
@@ -100,14 +146,14 @@ def _run_states(self, states_group: str, run_args: run_args_type):
         states = deploy_states[states_group]
         stages = run_args.stages
 
-        primary = local_minion_id()
+        primary = self._primary_id()
```

Comment:
that should be replaced everywhere, not only here
also:
- all direct calls to `provisioner.salt` API should be replaced with calls to parent (`Deploy`) class:
    - cmd_run -> self._cmd_run
    - local_minion_id() -> self._primary_id()
    - function_run -> self._function_run

The only missed thing in Deploy is StateFunExecuter. You can either:
- extend Deploy to provide that api to child classes
- use `self._function_run` since actually StateFunExecuter is just a wrepper over provisioner.salt.function_fun for `state.single` function

The general idea is support 2 cases of setup:
- remote setup so it can be combined with SetupProvisioner
- local setup so it might be run on a primary node locally

Deploy as a parent class takes care about that fro children classes.

Reply:
replaced all local function call with Deploy func calls.

Reply:
please check one last place